### ISSUE TITLE
Inaccurate Code Coverage on UDP server app

### ISSUE THREAD

**Issue by dmcnulla** [colszowka/simplecov#489]
I am testing a UDP server, not Rails, which seems to have very low code coverage (42%), but when I investigated I found [many lines that must have been exercised](https://coveralls.io/jobs/13494858/source_files/214774276) were marked as not covered. At first, I thought it was a coveralls problem. Then I tried codeclimate's coverage, the number was exactly the same. So I boiled it down to simplecov and also got the same coverage. I experimented with placement of lines of code in my env.rb (cukes) but the result was always the same. I tried moving the server start from the hooks to within the SimpleCov.start but it would not run, and all tests failed.

I'm sure based on the other issues with similar symptoms, there must be a workaround. I'd love to try things.

Submission checklist:
ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-linux] (thought I get this no matter which version of ruby I use)

Generated by simplecov v0.11.2 and simplecov-html v0.10.0 using features

coveralls (0.8.13)
simplecov (0.11.2)
simplecov-html (0.10.0)
codeclimate-test-reporter (0.5.0)

Rakefile:
```ruby
require 'bundler/gem_tasks'
require 'cucumber/rake/task'
require 'simplecov'
Cucumber::Rake::Task.new(:features) do |t|
  t.profile = 'default'
end
task default: [:features]
```
I also tried .simplecov, but no affect.

**Comment by colszowka** [comment]
@dmcnulla Please move the `require 'simplecov'`at the top of your cucumber env.rb, rake test tasks spawn subprocesses for actually running the tests and hence what happens there becomes invisible to simplecov if you load it in a Rakefile. See also #340.

Closing this, feel free to reopen if the issue persists.

**Comment by dmcnulla** [comment]
@colszowka I've moved require 'simplecov' to the very top of the env.rb, removed it from Rakefile, but I still get the same coverage percentage. Could it be anything else?

**Comment by colszowka** [comment]
Not sure exactly what might be wrong there without digging into the code, but I'd suspect that your test process (where simplecov is running) is separate from the one where your test server is running (although https://github.com/dmcnulla/ntp/blob/master/features/support/hooks.rb seems its the same, but I didn't look into how starting works in this case)

The issue is comparable to running a ruby webserver process, then hitting that server with HTTP request from a test suite - simplecov will not see the lines hit by the server process, just the querying logic in your test suite. You'd have to make sure you load simplecov in the server process, too, and set a distinct `SimpleCov.command_name 'Test Server'` test suite name for the merging then.

**Comment by dmcnulla** [comment]
You are right, it starts in the scenario hook. It ultimately starts in the [control class](https://github.com/dmcnulla/ntp/blob/master/lib/ntp/server/control.rb) like this:
```ruby
   def start port = DEFAULT_PORT
      fork do
         Process.setsid
         NTP::Server::Base.new(port || DEFAULT_PORT).start
      end
      read(@client_in, 5)
   end
```

I will try to make a clean branch using only simplecov (master branch uses coveralls) tonight in case you wanted to run it locally. That would be about 11-12 hours from now.

**Comment by dmcnulla** [comment]
I created a [new branch with just simplecov](https://github.com/dmcnulla/ntp/tree/simplecov).
[Results showing low coverage percentage](https://travis-ci.org/dmcnulla/ntp/jobs/122610063).
It's the same percentage as the coveralls & codeclimate attempts.

**Comment by dmcnulla** [comment]
I've been doing more experiments. I tried moving my start into the simplecov start (nothing would run, or I'd get 0 metrics). I tried coco, and the coverage results are the same as with simplecov.

I think I have a similar problem to rails caused by the forked process, but without it being similar enough to rails to use the same solution. I think that I need to drop the test coverage for now. Thank you for looking into the matter.
